examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHash.agent.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { INVALID_ANCHOR_HASH_INPUT } from '../api/err.messages';
+import { v4 as uuidv4 } from 'uuid';
+import { CCSMAnchorHash } from '../models/ccsmAnchorHash';
+
+@Injectable()
+export class CCSMAnchorHashAgent {
+  public throwErrorIfCCSMAnchorHashInputInvalid(
+    inputForProofVerification: string,
+  ): void {
+    if (!inputForProofVerification) {
+      throw new BadRequestException(INVALID_ANCHOR_HASH_INPUT);
+    }
+  }
+
+  public createNewCCSMAnchorHash(
+    ownerId: string,
+    document: string,
+  ): CCSMAnchorHash {
+    const hash = this.convertDocumentToHash(document);
+
+    return new CCSMAnchorHash(uuidv4(), ownerId, hash);
+  }
+
+  public verifyCCSMAnchorHash(
```

Comment:
@biscuitdey we have not talked about the prover system ...if we use Merkle trees, and the anchor hash is content addressable, then verification is done differently, as well as the inputs are different. Can you please specify the prover system and the anchor hash creation algorithm (note, you can steal that from IPFS)

Reply:
I have detailed the prover system in the Zero-knowledge interactions [document](https://docs.google.com/document/d/1mkO5oou0TWk6kVHbTB01dyj7wuNGTfR9w6PAxSgXdGI/edit?usp=sharing)